hibernate-core/src/main/java/org/hibernate/engine/transaction/internal/TransactionImpl.java

```diff
@@ -31,29 +32,36 @@
 	private final TransactionCoordinator transactionCoordinator;
 	private final ExceptionConverter exceptionConverter;
 	private final JpaCompliance jpaCompliance;
+	private final AbstractSharedSessionContract session;
 
 	private TransactionDriver transactionDriverControl;
 
 	public TransactionImpl(
 			TransactionCoordinator transactionCoordinator,
 			ExceptionConverter exceptionConverter,
-			JpaCompliance jpaCompliance) {
+			JpaCompliance jpaCompliance,
+			AbstractSharedSessionContract session) {
 		this.transactionCoordinator = transactionCoordinator;
 		this.exceptionConverter = exceptionConverter;
 		this.jpaCompliance = jpaCompliance;
+		this.session = session;
 
-		this.transactionDriverControl = transactionCoordinator.getTransactionDriverControl();
+		if ( session.isOpen() && transactionCoordinator.isActive() ) {
+			this.transactionDriverControl = transactionCoordinator.getTransactionDriverControl();
+		}
```

Comment:
else {
        LOG.debugf( "Transaction initiated on closed Session/SessionFactory" );
    }

?

Reply:
This is a question :)

Reply:
here I'm not sure about the message, the transaction is not really initiated .

Reply:
what about 
`LOG.debug("Transaction acquired on closed Session/EntityManager")`
?

Reply:
`LOG.debug( "TransactionImpl created on closed  Session/EntityManager")`?  That would match the phrasing of the other part of this method ("On TransactionImpl creation, JpaCompliance#isJpaTransactionComplianceEnabled == %s").

Reply:
+1